staging/src/k8s.io/legacy-cloud-providers/azure/azure_storage.go

```diff
@@ -36,25 +37,32 @@ const (
 
 // CreateFileShare creates a file share, using a matching storage account type, account kind, etc.
 // storage account will be created if specified account is not found
-func (az *Cloud) CreateFileShare(shareName, accountName, accountType, accountKind, resourceGroup, location string, protocol storage.EnabledProtocols, requestGiB int) (string, string, error) {
-	if resourceGroup == "" {
-		resourceGroup = az.resourceGroup
+func (az *Cloud) CreateFileShare(accountOptions *AccountOptions, shareOptions *fileclient.ShareOptions) (string, string, error) {
+	if accountOptions == nil {
+		return "", "", fmt.Errorf("account options is nil")
+	}
+	if shareOptions == nil {
+		return "", "", fmt.Errorf("share options is nil")
+	}
+	if accountOptions.ResourceGroup == "" {
+		accountOptions.ResourceGroup = az.resourceGroup
 	}
 
 	enableHTTPSTrafficOnly := true
-	if protocol == storage.NFS {
+	if shareOptions.Protocol == storage.NFS {
 		enableHTTPSTrafficOnly = false
 	}
-	account, key, err := az.EnsureStorageAccount(accountName, accountType, accountKind, resourceGroup, location, fileShareAccountNamePrefix, enableHTTPSTrafficOnly)
+
+	accountName, accountKey, err := az.EnsureStorageAccount(accountOptions, fileShareAccountNamePrefix, enableHTTPSTrafficOnly)
```

Comment:
`enableHTTPSTrafficOnly` is also part of `accountOptions` property

Reply:
done